doc/source/whatsnew/v2.0.0.rst

```diff
@@ -340,7 +340,7 @@ Datetimelike
 - Bug in :func:`to_datetime` was raising on invalid offsets with ``errors='coerce'`` and ``infer_datetime_format=True`` (:issue:`48633`)
 - Bug in :class:`DatetimeIndex` constructor failing to raise when ``tz=None`` is explicitly specified in conjunction with timezone-aware ``dtype`` or data (:issue:`48659`)
 - Bug in subtracting a ``datetime`` scalar from :class:`DatetimeIndex` failing to retain the original ``freq`` attribute (:issue:`48818`)
--
+- Bug in ``pandas.tseries.holiday.Holiday`` causing inconsistent return types for custom :class:`AbstractHolidayCalendar` instances  (:issue:`49075`)
```

Comment:
Could you expand when `causing inconsistent return types` occurs?

Reply:
How about
```
- Bug in ``pandas.tseries.holiday.Holiday`` where a half-open date interval causes inconsistent return types from :meth:`AbstractHolidayCalendar.holidays` (:issue:`49075`)
```

Reply:
That would be great, thanks!

Reply:
It was HILARIOUSLY fun to find this, despite what the commit history would indicate.